### DISCUSSION TITLE
Applying transform animations with floating-ui/vue

### DISCUSSION THREAD

**Discussion by DarthSaul** [floating-ui/floating-ui#2893] [Q&A]
Hello!

 

I'm trying to implement `transform`-based CSS animations on my tooltip content, so that I can create the effect of the tooltip content sliding out from the trigger. I'm starting with using the Vue built-in `<Transition`> component, but when I provide a `transform` property, I get the effect of the tooltip content first displaying at `left: 0, right: 0` on the document body, then sliding into the correct position. This behavior is expected, as by default, "the floating element is positioned using transform in the floatingStyles object." (from [floating-ui/vue docs](https://floating-ui.com/docs/vue#disabling-transform)).

In the same section of the docs linked above, I see the following tip:

> If you’d like to retain transform styles while allowing transform animations, create a wrapper, where the outermost node is the positioned one, and the inner is the actual styled element.

**Does anyone have an example of this implementation?** I've been trying lots of different things but getting nowhere :(

As the title indicates, I'm working in Vue.js with the `floating-ui/vue` library!

Thanks in advance!!

**Comment by NaXiDaXi** [comment]
Disable transform in the floatingStyles object

useFloating(reference, floating, {
  transform: false,
});

**Reply by alexmnv** [reply]
> Disable transform in the floatingStyles object

Opting out of `transform` is not a good solution if the tooltip needs to change positions, because using `transform` for positioning is more performant.

**Comment by ibuiltthis** [comment]
I gave up on getting transitions on transform to work because the floating element kept transitioning in form coordinates 0, 0.  In the end i just transitioned on opacity.

**Comment by atomiks** [comment]
The wrapper just means this. The topmost node is unstyled aside from positioning, while the inner one has the styles, including transform animations, etc.

```Vue
<div :style="floatingStyles">
  <div class="tooltip">
    content
  </div>
</div>
```

**Comment by Maarten-Dekker** [comment]
I encountered the same issue, but I found a working solution by wrapping the positioned floating element in a wrapper. Here is a dropdown component as reference:
```vue
<script setup>
import { autoUpdate, flip, shift, offset as spacing, useFloating } from '@floating-ui/vue';
import { shallowRef, useTemplateRef } from 'vue';

const props = defineProps({
    placement: {
        type: String,
        default: 'bottom-start',
    },
    offset: {
        type: Number,
        default: 6,
    },
});

const trigger = useTemplateRef('trigger');
const floating = useTemplateRef('floating');

const { floatingStyles } = useFloating(trigger, floating, {
    placement: props.placement,
    middleware: [spacing(props.offset), flip(), shift()],
    whileElementsMounted: autoUpdate,
});

const open = shallowRef(false);
</script>

<template>
    <button ref="trigger" @click.stop="open = !open">Dropdown</button>
    <Teleport to="body">
        <div ref="floating" :style="floatingStyles">
            <Transition
                enter-active-class="transition ease-out duration-200"
                enter-from-class="opacity-0 translate-y-1"
                enter-to-class="opacity-100 translate-y-0"
                leave-active-class="transition ease-in duration-150"
                leave-from-class="opacity-100 translate-y-0"
                leave-to-class="opacity-0 translate-y-1"
            >
                <div v-if="open" class="max-w-60 min-w-40 rounded-lg border border-gray-300 bg-white p-2 shadow-xs">
                    <slot />
                </div>
            </Transition>
        </div>
    </Teleport>
</template>
```
This approach ensures proper positioning and behavior. Hope this helps anyone facing the same problem!
